Examples/Nodify.Shapes/Canvas/CanvasViewModel.cs

```diff
@@ -0,0 +1,249 @@
+﻿using Nodify.Shapes.Canvas.UndoRedo;
+using Nodify.UndoRedo;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Input;
+
+namespace Nodify.Shapes.Canvas
+{
+    public class CanvasViewModel : ObservableObject
+    {
+        private readonly NodifyObservableCollection<ShapeViewModel> _shapes = new NodifyObservableCollection<ShapeViewModel>();
+        public IReadOnlyCollection<ShapeViewModel> Shapes => _shapes;
+        public NodifyObservableCollection<ShapeViewModel> SelectedShapes { get; } = new NodifyObservableCollection<ShapeViewModel>();
+
+        public NodifyObservableCollection<ICanvasDecorator> Decorators { get; } = new NodifyObservableCollection<ICanvasDecorator>();
+        public NodifyObservableCollection<ConnectionViewModel> Connections { get; } = new NodifyObservableCollection<ConnectionViewModel>();
+
+        private ShapeToolbarViewModel ShapeToolbar { get; } = new ShapeToolbarViewModel();
+        public CanvasToolbarViewModel CanvasToolbar { get; }
+
+        public ICommand MoveShapesStartedCommand { get; }
+        public ICommand MoveShapesCompletedCommand { get; }
+        public ICommand SelectShapesStartedCommand { get; }
+        public ICommand SelectShapesCompletedCommand { get; }
+        public ICommand ResizeShapeStartedCommand { get; }
+        public ICommand ResizeShapeCompletedCommand { get; }
+        public ICommand CreateConnectionCommand { get; }
+        public ICommand RemoveConnectionCommand { get; }
+        public ICommand DeleteSelectionCommand { get; }
+
+        public ICommand UndoCommand { get; }
+        public ICommand RedoCommand { get; }
+
+        public NodifyObservableCollection<UserCursorViewModel> Cursors { get; } = new NodifyObservableCollection<UserCursorViewModel>();
+        public IActionsHistory UndoRedo { get; } = ActionsHistory.Global;
```

Comment:
possibly a local/instanced one is more common

Reply:
I agree, but for simplicity, I would like to keep it global. Otherwise, I would have to pass the reference to the current instance everywhere I need it.